src/logical.pxi

```diff
@@ -43,6 +43,33 @@ cdef class LogicalType(ValueAdapter):
                 if inst is not None:
                     return inst
 
+import gzip
+from io import BytesIO
+
+
+cdef class GzipStringType(LogicalType):
+    """Logical type for gzip-compressed string values"""
+    logical_name = 'gzip-str'
+    underlying_types = (BytesType, )
+
+    @classmethod
+    def _for_type(cls, underlying: AvroType):
+        # Only apply this logical type to bytes type
+        return cls() if isinstance(underlying, BytesType) else None
+
+    cdef encode_value(self, value):
+        if not isinstance(value, str):
+            raise TypeError(f"Value to encode should be a string, got {type(value)}")
+        with BytesIO() as bytes_io:
+            with gzip.GzipFile(fileobj=bytes_io, mode='wb') as gzip_file:
```

Comment:
Should be possible to use gzip.compress / decompress directly, avoiding the need for bytesIO etc